lib/rucio/core/did.py

```diff
@@ -1452,8 +1465,12 @@ def touch_dids(dids, session=None):
     now = datetime.utcnow()
     try:
         for did in dids:
-            session.query(models.DataIdentifier).filter_by(scope=did['scope'], name=did['name'], did_type=did['type']).\
-                update({'accessed_at': did.get('accessed_at') or now}, synchronize_session=False)
+            row = session.query(models.DataIdentifier).filter_by(scope=did['scope'], name=did['name'], did_type=did['type']).one()
```

Comment:
I think a conditional update would be better here. Something like https://github.com/rucio/rucio/blob/master/lib/rucio/core/replica.py#L1467
In that case you don't need to read, calculate, update but can just du an update with access_cnt+1